### PR TITLE
feat: add metadata & hash for metrics

### PR DESCRIPTION
**Pull request by hengfeiyang**
- [x] add hash for series
- [x] add metadata for schema

The `Metadata` define:

```Rust
pub struct Metadata {
    pub metric_type: MetricType,
    pub metric_family_name: String,
    pub help: String,
    pub unit: String,
}
```

### CODE REVIEW

File: src/meta/prom.rs [hunk 1]
```diff
@@ -32,23 +40,56 @@ pub struct ClusterLeader {
...
+pub enum MetricType {
+    UNKNOWN,
+    COUNTER,
+    GAUGE,
+    HISTOGRAM,
+    GAUGEHISTOGRAM,
+    SUMMARY,
+    INFO,
+    STATESET,
+}
```
Comment (vvv) [inline src/meta/prom.rs:53]:
```suggestion
// Cf. https://github.com/prometheus/prometheus/blob/f5fcaa3872ce03808567fabc56afc9cf61c732cb/model/textparse/interface.go#L106-L119
#[derive(Clone, Debug, Serialize, Deserialize)]
#[serde(rename_all = "UPPERCASE")]
pub enum MetricType {
    Counter,
    Gauge,
    Histogram,
    GaugeHistogram,
    Summary,
    Info,
    StateSet,
    Unknown,
}
```
```

**Review by oasisk — APPROVED** [review]

File: src/meta/prom.rs [hunk 2]
```diff
@@ -32,23 +40,56 @@ pub struct ClusterLeader {
...
+    pub fn as_str_name(&self) -> &'static str {
+        match self {
+            MetricType::UNKNOWN => "UNKNOWN",
+            MetricType::COUNTER => "COUNTER",
+            MetricType::GAUGE => "GAUGE",
+            MetricType::HISTOGRAM => "HISTOGRAM",
+            MetricType::GAUGEHISTOGRAM => "GAUGEHISTOGRAM",
+            MetricType::SUMMARY => "SUMMARY",
+            MetricType::INFO => "INFO",
+            MetricType::STATESET => "STATESET",
```
Comment (vvv) [inline src/meta/prom.rs:80]: Use [`strum`](https://lib.rs/crates/strum).

File: src/service/schema.rs [hunk 3]
```diff
@@ -287,24 +288,18 @@ pub async fn stream_schema_exists(
...
-        return schema_chk;
-    }
-    if let Some(value) = meta.get("settings") {
+    if let Some(value) = schema.metadata().get("settings") {
         let settings: json::Value = json::from_slice(value.as_bytes()).unwrap();
         if settings.get("partition_keys").is_some() {
             schema_chk.has_partition_keys = true;
         }
     }
+    if schema.metadata().get("metadata").is_some() {
```
Comment (vvv) [inline src/service/schema.rs]:
```suggestion
    if schema.metadata().contains("metadata") {
```

**Review by vvv — APPROVED** [review]

File: src/service/metrics/mod.rs [hunk 4]
```diff
@@ -20,7 +20,7 @@ use crate::meta::prom::Metadata;
 pub mod prom;
 
 pub async fn get_prom_metadata_from_schema(schema: &Schema) -> Option<Metadata> {
-    let metadata = schema.metadata.get("metadata")?;
+    let metadata = schema.metadata.get("prom_metadata")?;
```
Comment (vvv) [inline src/service/metrics/mod.rs:23]: [optional] Consider using a constant instead of string literal.

File: src/service/schema.rs [hunk 5]
```diff
@@ -297,7 +297,7 @@ pub async fn stream_schema_exists(
             schema_chk.has_partition_keys = true;
         }
     }
-    if schema.metadata().get("metadata").is_some() {
+    if schema.metadata().get("prom_metadata").is_some() {
```
Comment (vvv) [inline src/service/schema.rs:300]:
```suggestion
    if schema.metadata().contains("prom_metadata") {
```

File: src/meta/prom.rs (hunk 1 above)
Comment (hengfeiyang) [reply src/meta/prom.rs:53]: this define i copied from Prometheus, i think it is ok.

File: src/service/schema.rs (hunk 3 above)
Comment (hengfeiyang) [reply src/service/schema.rs]: updated

**Pull request merged**